pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -227,7 +227,11 @@ func (t *ResolvedPipelineRunTask) parentTasksSkip(facts *PipelineRunFacts) bool
 	stateMap := facts.State.ToMap()
 	node := facts.TasksGraph.Nodes[t.PipelineTask.Name]
 	for _, p := range node.Prev {
-		if stateMap[p.Task.HashKey()].Skip(facts) {
+		parentTask := stateMap[p.Task.HashKey()]
+		if parentTask.Skip(facts) {
+			if facts.ScopeWhenExpressionsToTask && parentTask.PipelineTask.WhenExpressions != nil {
+				continue
+			}
```

Comment:
im confused what the purpose of this task is based on the name `parentTasksSkip` - what is this function's job? if there was any way you'd be willing to take on a bit of scope creep and at least add a docstring if not also rename this function id be very appreciative 🙏 

(i cant tell if this functions job is to say: should this task be skipped since it's parent task was skipped, or maybe: was the parent task skipped, or something else)

Reply:
i think it would also help to have some unit tests for just this function - often a sign that this unexported function deserves to be in its own package somewhere and exported :D

when i look at pipelinerunresolution_test, it seems like the behavior is what i would expect, but when i look at this logic in this function i dont understand how that's being accomplished - some unit test directed at just this function might help

Reply:
> when i look at pipelinerunresolution_test, it seems like the behavior is what i would expect, but when i look at this logic in this function i dont understand how that's being accomplished - some unit test directed at just this function might help

sounds good, added unit tests for that function (checking whether to skip because of parent tasks)

Reply:
thanks @jerop, makes a lot more sense to me now!! 🙏